### ISSUE TITLE
cmakelint warnings

### ISSUE THREAD

**Issue by dfandrich** [curl/curl#14580]
### I did this

I tried running cmakelint on the curl sources with the command `cmakelint --spaces=2 --linelength=132 CMake/* $(find . -type f -name CMake\*) ` and it showed [these warnings](https://github.com/user-attachments/files/16646644/cmakelint.txt).

I don't have an opinion on the veracity of these warnings (@vszakats probably would), but if they look useful, this would be easy to add to a CI job.

### I expected the following

cmakelint to be warning-free.

### curl/libcurl version

8.10.0-DEV

### operating system

Linux

**Comment by vszakats** [comment]
Nice!

**Comment by vszakats** [comment]
Pushed some fixes to this existing PR #14579

1. I was just wondering about the naming of Find* modules. It's a bit all over the place. Also CMake built-in ones are.
`cmakelint` suggests full uppercase component names after `Find`. I'm fine with this, but as this `cmakelint` Issue [notes](https://github.com/cmake-lint/cmake-lint/issues/24), CMake itself [doesn't suggest](https://cmake.org/cmake/help/latest/manual/cmake-developer.7.html#standard-variable-names) such convention and I also cannot see it confirmed "in the wild".

I'd probably just do capital and then use whatever the component is named, and leave most existing uppercased ones as-is, changing only these:
```
FindLibPSL  -> FindLibpsl
FindLibSSH2 -> FindLibssh2
FindQUICHE  -> FindQuiche
Findrustls  -> FindRustls
```

Either way it will need a separate PR just for that. But first what scheme should we follow here, if any?

2. As for the "weird indentation" issues: It caught two that I was aware of (fixed now!), but the rest I'm conflicted with. `cmakelint` wants to indent multiline `if(` expression to the 2-char levels and calls it weird when aligned with the preceding `if(`. This recommendation IMO reduce readability for these expressions.

Found these these two solution:
- avoid multiline expressions. By unfolding or splitting to nested `if()`s. This is also weird, and may not be applicable to all expressions without reducing readability.
- suppress the warning with `--filter=-whitespace/indent`. This looks harsh, but this option doesn't seem to catch real indentation issues anyway. (tried by unindenting, overindenting instructions under a `function` and `if`, neighter triggerring warnings.

**Comment by dfandrich** [comment]
I noticed that multiline indentation issue as well. That might be worth
bringing upstream as a bug, since indenting while ignoring the parenthesis
is definitely a step down in readability.

Even if half the checks are disabled, it still leaves half the checks in
effect, so it's still a gain in consistency.

**Comment by vszakats** [comment]
Absolutely, I think it's a nice candidate for a new CI job. Would you want to add one?

**Comment by vszakats** [comment]
However I try to extract some "packagename" convention or recommendation from the internet, there is hardly any. CMake authors cannot or are unwilling to recommend anything tangible.

Even the often seen capital-case rule is wrong; simply there is no package named `Libssh2` or `Libidn2`, so naming them just for the sake of having a filename `FindCapital.cmake` seems ...odd. All the while all internal variable references are `LIBSSH2_*` or `LIBIDN2_*` in uppercase.

Almost nobody uses a `Findlowercase.cmake` styled name, even though that would make most sense to match how these packages are named in every existing package manager and by most projects refer to themselves.

Perhaps one clue could be what project name each dependency uses in their CMake build files (if they have them). For `libssh2` this would be `libssh2` in lowercase, same for `brotli`, `libssh`, `nghttp2`, `nghttp3`, `ngtcp2`, `wolfssl`, `zlib`, `zstd`. Exceptions are: `c-ares` (lowercase with dash), `Mbed TLS`, `LibreSSL` (doesn't need a Find module).

```
lower               mixed               upper
Findbearssl.cmake   FindBearSSL.cmake   FindBEARSSL.cmake
Findbrotli.cmake    FindBrotli.cmake    FindBROTLI.cmake
Findcares.cmake     FindCARES.cmake     FindCARES.cmake
Findgsasl.cmake     FindGSASL.cmake     FindGSASL.cmake
Findgss.cmake       FindGSS.cmake       FindGSS.cmake
Findlibidn2.cmake   FindLibidn2.cmake   FindLIBIDN2.cmake
Findlibpsl.cmake    FindLibpsl.cmake    FindLIBPSL.cmake
Findlibssh.cmake    FindLibssh.cmake    FindLIBSSH.cmake
Findlibssh2.cmake   FindLibssh2.cmake   FindLIBSSH2.cmake
Findlibuv.cmake     FindLibuv.cmake     FindLIBUV.cmake
Findmbedtls.cmake   FindMSH3.cmake      FindMBEDTLS.cmake
Findmsh3.cmake      FindMbedTLS.cmake   FindMSH3.cmake
Findnettle.cmake    FindNGHTTP2.cmake   FindNETTLE.cmake
Findnghttp2.cmake   FindNGHTTP3.cmake   FindNGHTTP2.cmake
Findnghttp3.cmake   FindNGTCP2.cmake    FindNGHTTP3.cmake
Findngtcp2.cmake    FindNettle.cmake    FindNGTCP2.cmake
Findquiche.cmake    FindQuiche.cmake    FindQUICHE.cmake
Findrustls.cmake    FindRustls.cmake    FindRUSTLS.cmake
Findwolfssh.cmake   FindWolfSSH.cmake   FindWOLFSSH.cmake
Findwolfssl.cmake   FindWolfSSL.cmake   FindWOLFSSL.cmake
Findzstd.cmake      FindZstd.cmake      FindZSTD.cmake
```

**Comment by vszakats** [comment]
more tests later...

The filename of the Find modules must match the name passed to `find_package()`. This isn't obvious at first because it's not document, and it depends on the filesystem's case-sensitivity. E.g. on macOS (and Windows) they are matched case-insensitvely.

That name also must match the one passed to `find_package_handle_standard_args()` in the Find module itself, to avoid a CMake warning.

CMake also picks up a 'cmake-config' file with this name. According to [docs](https://cmake.org/cmake/help/latest/command/find_package.html#search-modes), it looks for: `<lowercasePackageName>-config.cmake`, `<PackageName>Config.cmake`, and `<lowercasePackageName>-config-version.cmake`.

Meaning it may be useful to match the casing of package provided 'cmake-config', if there is one.

I'm shifting towards the ambiguous and arbitrary, but probably more "CMake-like" _mixed case_ naming, with a preference to match the package convention, and falling back to Capitalized otherwise.

**Comment by dfandrich** [comment]
Sure, I can add a job for this.

**Comment by vszakats** [comment]
@dfandrich Sounds good!

Opened a PR with the renames mentioned above:
https://github.com/curl/curl/pull/14601

I think this describes the convention that I'm aiming at with package names:
Always start with an uppercase case and follow with lowercase, unless there
is a clear preference for a stylized name. (E.g. the project itself uses it that
way with a matching `<Name>Config.cmake` file, we use it that way elsewhere,
or the name is an acronym.
